ruby.rb

```diff
@@ -3,49 +3,51 @@ class ArticlesController < ApplicationController
 
   def index
     @articles = Article.all.includes(:user)
+    articles_type
+    boundary = articles_boundary(0, 20)
+    offset = boundary[offset]
+    limit = boundary[limit]
+    @articles = @articles.order(created_at: :desc).offset(offset).limit(limit)
+  end
 
+  def articles_type
     if params[:tag].present?
```

Comment:
The controller should not be concerned with this logic. Explicitly checking the params in a controller is very unnecessary. This also violates Open Closed principle. Have you heard of query objects? This is a candidate for query objects.

Imagine if we had something like: `ArticleQuery.new(params)`. And then it responds to a `results` method. The details of how these params are processed are hidden within this class.

Reply:
Also, I want you to read the refactoring pattern called `Refactor Conditional with Polymorphism`. See how that can help you within the `ArticleQuery` class to remove all this unnecessary conditionals.

Also, this new class you create should be nested under `app/queries/article_query.rb`